### PR TITLE
Fix lightbulb issues

### PR DESCRIPTION
**Pull request by rchande**
The move to SuggestedActionsSource2 introduced 2 issues tracked by https://github.com/dotnet/roslyn/issues/19772. This change fixes both:
1. The "refactoring" lightbulb would sometimes be available but disappear if clicked on
2. QuickInfo for error squiggles would not show a lightbulb for available fixes unless the caret was within the squiggle's span.

Tag @dotnet/roslyn-ide for review

### CODE REVIEW

File: src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs [hunk 1]
```diff
@@ -891,33 +891,32 @@ private void OnSuggestedActionsChanged(Workspace currentWorkspace, DocumentId cu
...
-                        var document = range.Snapshot.GetOpenDocumentInCurrentContextWithChanges();
-                        using (var linkedTokenSource = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken))
+                        var linkedToken = linkedTokenSource.Token;
+
+                        var errorTask = Task.Run(
+                            () => GetFixLevelAsync(provider, document, range, linkedToken), linkedToken);
+
+                        var selection = await GetSpanAsync(range).ConfigureAwait(false);
+                        Task<string> refactoringTask = Task.FromResult((string)null);
+                        if (selection != null && requestedActionCategories.Contains(PredefinedSuggestedActionCategoryNames.Refactoring))
```
Comment (gundermanc) [inline src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs]: Does this code do the right thing if we ask for a different set of requested actions in the future? It's not an issue yet, but I could see the platform team wanting to add a new entry point to lightbulb in the future that requests only errors, or something similar.

Comment (rchande) [inline src/EditorFeatures/Core/Implementation/Suggestions/SuggestedActionsSource.cs]: You're correct--it only respects the presence of the refactoring category. Looking at some of the PredefinedSuggestedActionCategoryNames, there are categories like "StyleFix" that currently don't map onto our severity model. For that reason, I'd prefer to not attempt to fully plumb through all the remaining category names for now (given that respecting refactorings fixes the bug :)).

**Comment by rchande** [comment]
retest windows_build_correctness_prtest

**Comment by rchande** [comment]
**Customer scenario**

Lightbulb appears offering refactorings but diappears when the customer clicks on it.
Customer mouses over an error squiggle but the lightbulb with fix does not appear unless they put the caret within the span of the error squiggle.

**Bugs this fixes:**

https://github.com/dotnet/roslyn/issues/19772


**Workarounds, if any**

Refactoring issue: none. Squiggle--move caret into squiggle range.

**Risk**

Low--makes a null check more specific.

**Performance impact**

Low--we will compute refactorings less often.

**Is this a regression from a previous update?**

Yes, this is was introduced when we moved the ISuggestedActionSource2.

**Root cause analysis:**

Assumed the editor would not request specific action categories

**How was the bug found?**

Dogfooding.

**Comment by rchande** [comment]
Tag @MattGertz for ask mode for 15.4P2

**Comment by rchande** [comment]
retest windows_debug_vs-integration_prtest

**Pull request merged**